src/calendar/use-calendar-grid-keyboard-navigation.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import React from 'react';
+import { CalendarProps } from './interfaces';
+import { moveNextDay, movePrevDay, moveNextWeek, movePrevWeek } from './utils/navigation';
+import { KeyCode } from '../internal/keycode';
+import { isSameMonth } from 'date-fns';
+
+export default function useCalendarGridKeyboardNavigation({
```

Comment:
Maybe a philosophical question but is it really a hook if it uses no React hooks? (https://react.dev/reference/react/hooks)

This just feels like a utility function and therefore the nomenclature (`use*`) is a bit misleading.

Reply:
Went the full React way and added a `useCallback`...

Reply:
It is not required to `useCallback` to satisfy philosophy of the hooks. Here is an example, similar to yours: https://github.com/cloudscape-design/components/blob/4f7fc2c307de932b06c207931b3a76d4cf994364/src/app-layout/utils/use-keyboard-events.ts#L24

Worked well so far.

---

If you strongly dislike use-less hooks, you can make it a plain function:

```js
import {handleGridNavigation} from './handle-grid-navigation'

onGridKeyDownHandler={event => handleGridNavigation(...)}
```

Reply:
Another note, since we have `switch (event.keyCode)` in way too many places, is this the time to introduce a utility already?

```js
handleKeyNavigation(event, {
   onActivate: () => /*enter or space*/
   onUp: () => {},
   onDown: () => {},
   onLeft: () => {},
   onRight: () => {},
})
```

Reply:
Thanks both for the feedback. So I think we have two main questions here:

- Hook or no hook: since this seems to be an essentially philosophical question, and as Boris pointed out, we have other examples of this, I am inclined to keep it as a hook. And related to this question, the sub-question about `useCallback`: I could not measure any measurable performance impact of using it, and couldn't detect re-renders by not using it, so I removed it.
- A utility for keyboard navigation sounds like a good idea. I can explore it but I'd say in a different PR; then it can be integrated here.